### PR TITLE
chore: add letter validation for password field and tests

### PR DESCRIPTION
**Pull request by cbernat**
- Added letter validator
- Added tests for letter, number and empty password
- Changed digit to number as requested by mkt
image

### CODE REVIEW

**Comment by andresmoschini** [comment]
Nice work! thanks for the tests!

As we talked in a chat, @cbernat will add 3 more test scenarios:

* `11111` => _short and letter_
* `!!!!` => _short, digit and letter_
* `!!!!!!!!!!` => _digit and letter_

**Comment by GusBaamonde** [comment]
Great job @cbernat!!

File: src/validations.ts [hunk 1]
```diff
@@ -26,25 +26,39 @@ export function validateRequiredField(
...
+  }
   if (charError && digitError) {
     return { charLength: true, digit: true };
   }
+  if (charError && letterError) {
+    return { charLength: true, letter: true };
+  }
+  if (digitError && letterError) {
+    return { digit: true, letter: true };
+  }
```
Comment (andresmoschini) [inline src/validations.ts]: Is it not possible to simplify this code in the following way:

```js
if (charError || digitError || letterError) {
    return { charLength: charError, digit: digitError, letter: letterError };
}
```

Comment (andresmoschini) [reply src/validations.ts]: In place of: 

```js
  if (charError && digitError && letterError) {
    return { charLength: true, digit: true, letter: true };
  }
  if (charError && digitError) {
    return { charLength: true, digit: true };
  }
  if (charError && letterError) {
    return { charLength: true, letter: true };
  }
  if (digitError && letterError) {
    return { digit: true, letter: true };
  }
  if (charError) {
    return { charLength: true };
  }
  if (digitError) {
    return { digit: true };
  }
  if (letterError) {
    return { letter: true };
  }
```

Comment (cbernat) [reply src/validations.ts]: Changed.

**Pull request merged**